### PR TITLE
Don't post case ids if there is no filter applied

### PR DESCRIPTION
**Pull request by zhx828**

### CODE REVIEW

File: app/_components/QueryByGeneUtil.js [hunk 1]
```diff
@@ -19,25 +19,35 @@ window.QueryByGeneUtil = (function() {
...
       var _arr = [];
-      _.each(selectedCases, function(_obj) {
-        var _studyId = _obj.id;
-        _.each(_obj.samples, function(_sampleId) {
-          _arr.push(_studyId + ":" + _sampleId);
+      var formOps = {
+        cancer_study_list: studyIds
+      };
+
+      if (_.isObject(selectedCases)) {
```
Comment (kalletlak) [inline app/_components/QueryByGeneUtil.js]: could you make sure `selectedCases` is null when no filters applied.

File: app/scripts/main.js [hunk 2]
```diff
@@ -658,13 +658,13 @@ window.iViz = (function(_, $, cbio, QueryByGeneUtil, QueryByGeneTextArea) {
 
       if (_self.cohorts_.length === 1) { // to query single study
         if (QueryByGeneTextArea.isEmpty()) {
-          QueryByGeneUtil.toMainPage(_self.cohorts_[0], _self.stat().studies);
+          QueryByGeneUtil.toMainPage(_self.cohorts_[0], vm_.hasfilters ? _self.stat().studies : undefined);
```
Comment (kalletlak) [inline app/scripts/main.js]: Question, should `cancer_study_list` be input studies or selected studies after applying filters?

File: app/_components/QueryByGeneUtil.js [hunk 3]
```diff
@@ -19,25 +19,35 @@ window.QueryByGeneUtil = (function() {
...
-      });
-      submitForm(window.cbioURL + 'index.do', {
-        cancer_study_list: selectedCases.map(function(x) {return x.id}),
-        cancer_study_id: 'all',
-        case_set_id: -1,
-        case_ids: _arr.join('+')
-      });
+
+        formOps = {
+          cancer_study_list: selectedCases.map(function(t) {
```
Comment (kalletlak) [inline app/_components/QueryByGeneUtil.js:42]: I think `selectedCases.map(study => study.id) would be cleaner

Comment (zhx828) [reply app/_components/QueryByGeneUtil.js:42]: Are we doing es6 going forward?

File: app/scripts/main.js (hunk 2 above)
Comment (zhx828) [reply app/scripts/main.js]: selected studies after applying filters

Comment (kalletlak) [reply app/scripts/main.js]: Sorry, I got this. If no filters applied first parameter would be used else studies from stat would be used.

Comment (kalletlak) [reply app/scripts/main.js]: instead of passing `undefined` and checking it, why not just pass `_self.stat().studies` and `hasfilters` flag.

**Review by kalletlak — APPROVED** [review]

**Pull request merged**